### PR TITLE
Enhancement/4182 refresh tokens on sitekit pages

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #4182

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @eugene-manuilov – this is very close, just a small change left to make regarding the change to the hook used in one instance.

File: includes/Core/Authentication/Authentication.php [hunk 1]
```diff
@@ -426,8 +430,22 @@ function ( $user_id ) {
 			$this->get_oauth_client()->refresh_token();
 		};
 
-		add_action( 'admin_init', $maybe_refresh_token );
-		add_action( 'heartbeat_tick', $maybe_refresh_token );
+		add_action(
+			'admin_init',
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Core/Authentication/Authentication.php]: :man_facepalming: Good catch, @aaemnnosttv. Updated.

**Review by felixarntz — APPROVED** [review]
LGTM! Using `current_screen` instead of `admin_init` should work as expected. Even in the unlikely scenario that it might be invoked again, we won't refresh the token again since that would have already happened before.

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks @eugene-manuilov 👍

**Pull request merged**